### ISSUE TITLE
Make daily DB dumps available for Docker Cloud hosted sites.

### ISSUE THREAD

**Issue by cogat** [ic-labs/django-icekit#91]
TLDR:

jamesmurty [12:24 PM] Star this message
My suggestion: a scheduled job to dump the DB and upload to a private S3 bucket + a way to trigger a dump on-demand. Expose the S3 listing via the site's admin, which is already autenticated, and the site itself provides a pre-signed URL to download the dumps.

UPDATE: Implement as an ICEkit app, or a standalone django app (it doesn't really require any ICEkit features).

LONG VERSION:

tailee [12:08 PM]
when running on docker cloud and RDS, for getting local DB dumps from production/staging servers, i had planned to do something like... 1) run a btsync service on each node that dumps all databases on RDS and syncs them to the office mac mini and/or whoever wants them...; or 2) add a transfer.sh alias/script to the Dockerfile that dumps and encrypts the DB and uploads it via transfer.sh, giving you a URL to download it and decompress it...; or 3) we could add that as a feature to ICEkit admin for superusers... click a link, it spawns a celery task that dumps the DB, encrypts and uploads it via transfer.sh, and gives you a command to paste to download/decrypt it?

ryanstuart [12:11 PM] Star this message
I like the celery idea.

jamesmurty [12:13 PM] Star this message
Or we could update Deployo to do it more reliably? That works pretty well on other sites and doesn't involve an intermediate service

tailee [12:13 PM] Star this message
deployo wont work with docker cloud... docker cloud countainers dont run SSH

ryanstuart [12:14 PM] Star this message
And they really shouldn’t.

tailee [12:15 PM] Star this message
we could also encrypt/upload to our own S3 media instead of transfer.sh... but transfer.sh is pretty cool... files up to 10GB i think

jamesmurty [12:16 PM] Star this message
Fair enough, I guess we will need a lot more tooling with Docker-based sites for things that cannot be done the old way

ryanstuart [12:16 PM] Star this message
Transfer.sh is cool.

[12:16] Star this message
Never seen it before.

jamesmurty [12:17 PM] Star this message
I personally don't like the idea of shunting our DB dumps through a third-party service, but maybe it's just me

tailee [12:18 PM] Star this message
i know where you are coming from... so using our own S3 storage would be fine... we could still display a one liner to fetch and decrypt them just the same

ryanstuart [12:18 PM] Star this message
The really bone headed response to that would be that the DBs are already a 3rd party service :laughing:

tailee [12:19 PM] Star this message
S3 storage is public-readable though, and doesnt expire (unless we also do some work to remove old dumps)... transfer.sh would be public readable, but would expire automatically after 2 weeks

ryanstuart [12:20 PM] Star this message
I was just about to say, to be ultra secure, you shouldn’t use a public bucket. bucket.

jamesmurty [12:20 PM] Star this message
S3 storage is only public-readable if you make it so

ryanstuart [12:20 PM] Star this message
How about a celery beat job that does what you just said and maintains 14 days of history, and a fabfile (or whatever) to fetch from S3 using boto?

tailee [12:21 PM] Star this message
if we didnt use a public bucket, we'd need to provide the S3 bucket credentials (basic auth?) in the on-liner to fetch and decrypt?

ryanstuart [12:21 PM] Star this message
So instead of a button and a 1 liner, a dump is always available with a simple command.

[12:22] Star this message
Guaranteed to be no older than 24 hours.

jamesmurty [12:22 PM] Star this message
I've noticed a few comments about our publicly visible S3 setup. I think we're probably doing it wrong. For private (as they are by default) S3 objects, you either download them with an authenticated client or get a signed URL

ryanstuart [12:22 PM] Star this message
Or, even better, expose it as an API endpoint so you can run the command locally.

tailee [12:23 PM] Star this message
i'd rather have a list of available dumps in the admin with one liners to paste or links to click to download the files, than having to have Fabric installed into a venv and and the fab file available to fetch them

[12:24] Star this message
@jamesmurty: if we can upload the dumps to a private bucket and provide a signed URL that we can fetch with curl, cool...

jamesmurty [12:24 PM] Star this message
My suggestion: a scheduled job to dump the DB and upload to a private S3 bucket + a way to trigger a dump on-demand. Expose the S3 listing via the site's admin, which is already autenticated, and the site itself provides a pre-signed URL to download the dumps.

tailee [12:25 PM] Star this message
yep that sounds good

jamesmurty [12:26 PM] Star this message
Cool. Now that the hard work is all done...
